src/pubkey.cpp

```diff
@@ -166,6 +167,27 @@ static int ecdsa_signature_parse_der_lax(const secp256k1_context* ctx, secp256k1
     return 1;
 }
 
+XOnlyPubKey::XOnlyPubKey(Span<const unsigned char> in)
+{
+    assert(in.size() == 32);
+    std::copy(in.begin(), in.end(), m_keydata.begin());
+}
+
+bool XOnlyPubKey::VerifySchnorr(const uint256& msg, Span<const unsigned char> sigbytes) const {
+    if (sigbytes.size() != 64) return false;
```

Comment:
I don't get `feature_taproot.py` failure for :

```
diff --git a/src/pubkey.cpp b/src/pubkey.cpp
index 111f9c10e..0163f61a2 100644
--- a/src/pubkey.cpp
+++ b/src/pubkey.cpp
@@ -174,7 +174,7 @@ XOnlyPubKey::XOnlyPubKey(Span<const unsigned char> in)
 }
 
 bool XOnlyPubKey::VerifySchnorr(const uint256& msg, Span<const unsigned char> sigbytes) const {
-    if (sigbytes.size() != 64) return false;
+    if (sigbytes.size() != 64) return true;
     if (msg.size() != 32) return false;
     secp256k1_xonly_pubkey pubkey;
     if (!secp256k1_xonly_pubkey_parse(secp256k1_context_verify, &
```

And same for next check on message size.

Reply:
I've changed the `sigbytes` length check to an assert. The `msg` length check was just stupid; it's a `uint256`, its length is always 32.